### PR TITLE
Added a "constant" option to remove or rename enum constant

### PR DESCRIPTION
**Pull request by GeReV**
See update to README.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -79,6 +79,22 @@ event.status_available?     # when prefix true
...
+    enum :status, [:available, :canceled, :completed]
+    enum :without_const, [:foo, :bar], :constant => false
+    enum :custom_name, [:a, :b, :c], :constant => :special_name
+end
+
+event::STATUSES             # returns [:available, :canceled, :completed]
+event::WITHOUT_CONST        # raises NameError
+event::SPECIAL_NAME         # returns [:a, :b, :c]
+event::CUSTOM_NAME          # raises NameError
+```
```
Comment (yonbergman) [inline Readme.md]: 👍 for updating the readme

Comment (yonbergman) [inline Readme.md]: maybe note that the special name won't be pluralized

File: lib/enumify/model.rb [hunk 2]
```diff
@@ -18,7 +18,12 @@ def enumify(parameter, vals=[], opts={})
           prefix = "#{opts[:prefix].to_s}_"
         end
 
-        const_set("#{paramater_string.pluralize.upcase}", vals)
+        constant = opts.fetch(:constant, true)
```
Comment (yonbergman) [inline lib/enumify/model.rb]: nice

File: lib/enumify/model.rb [hunk 3]
```diff
@@ -18,7 +18,12 @@ def enumify(parameter, vals=[], opts={})
...
+        if constant
+          const_name = constant === true ? paramater_string.pluralize : constant.to_s
+
+          const_set("#{const_name.upcase}", vals)
```
Comment (yonbergman) [inline lib/enumify/model.rb]: why do we need to wrap in string?

**Review by yonbergman — APPROVED** [review]

**Comment by yonbergman** [comment]
some test fail :(

File: lib/enumify/model.rb (hunk 3 above)
Comment (GeReV) [inline lib/enumify/model.rb]: Good point. I think it was also there previously for no good reason.

**Pull request merged**